(dp1
S'output'
p2
S"<class 'novaclient.exceptions.NotFound'> No Server matching {'id': ''}. (HTTP 404)"
p3
sS'layer'
p4
S'/home/archer/Documents/web2py/applications/baadal/controllers/action.py'
p5
sS'code'
p6
S'import Baadal\nimport json\nimport ConfigParser\n# work around to stop stupid editors from complaining about undeclared \'request\'\nif False:\n    request = dict()\nconfig = ConfigParser.ConfigParser()\nconfig.read(\'/etc/baadal/baadal.conf\')\nauthurl = config.get(\'auth\',\'authurl\')\ntenant = config.get(\'auth\',\'tenant\')\npassword = config.get(\'auth\',\'password\')\nusername = config.get(\'auth\', \'username\')\n\ndef __do(action, vmid):\n    \n    conn = Baadal.Connection(authurl, tenant, username, password)\n    #conn = Baadal.Connection("http://10.237.23.178:35357/v2.0", "admin", "admin", "baadal")\n    if conn:\n        vm = conn.findBaadalVM(id=request.vars.vmid)\n        if vm:\n            if action == \'start\':\n                vm.start()\n            elif action == \'shutdown\':\n                vm.shutdown()\n            elif action == \'pause\':\n                vm.pause()\n            elif action == \'reboot\':\n                vm.reboot()\n            elif action == \'delete\':\n                vm.delete()\n        conn.close()\n        return json.dumps(dict(status=\'success\', code=\'OK\'))\n    else:\n        conn.close()\n        return dict(res = json.dumps(dict(status=\'failure\')))\n\ndef start():\n    return __do(\'start\', request.vars.vmid)\n    \ndef shutdown():\n    return __do(\'shutdown\', request.vars.vmid)\n    \ndef pause():\n    return __do(\'pause\', request.vars.vmid)\n    \ndef reboot():\n    return __do(\'reboot\', request.vars.vmid)\n\ndef delete():\n    return __do(\'delete\', request.vars.vmid)\n\nresponse._vars=response._caller(pause)\n'
p7
sS'snapshot'
p8
(dp9
sS'traceback'
p10
S'Traceback (most recent call last):\n  File "/home/archer/Documents/web2py/gluon/restricted.py", line 227, in restricted\n    exec ccode in environment\n  File "/home/archer/Documents/web2py/applications/baadal/controllers/action.py", line 52, in <module>\n  File "/home/archer/Documents/web2py/gluon/globals.py", line 412, in <lambda>\n    self._caller = lambda f: f()\n  File "/home/archer/Documents/web2py/applications/baadal/controllers/action.py", line 44, in pause\n    return __do(\'pause\', request.vars.vmid)\n  File "/home/archer/Documents/web2py/applications/baadal/controllers/action.py", line 19, in __do\n    vm = conn.findBaadalVM(id=request.vars.vmid)\n  File "applications/baadal/modules/Baadal.py", line 373, in findBaadalVM\n    baadalVM = self.__conn[\'nova\'].servers.find(**kwargs)\n  File "/home/archer/Documents/python-openstack/local/lib/python2.7/site-packages/novaclient/base.py", line 200, in find\n    raise exceptions.NotFound(404, msg)\nNotFound: No Server matching {\'id\': \'\'}. (HTTP 404)\n'
p11
s.